IPython/sphinxext/ipython_directive.py

```diff
@@ -358,9 +359,9 @@ def process_image(self, decorator):
         source_dir = self.source_dir
         saveargs = decorator.split(' ')
         filename = saveargs[1]
-        # insert relative path to image file in source
-        outfile = os.path.relpath(os.path.join(savefig_dir,filename),
-                    source_dir)
+        # insert relative path to image file in source (as absolute path for Sphinx)
+        outfile = '/' + os.path.relpath(os.path.join(savefig_dir,filename),
```

Comment:
I believe `/` is right, because it's effectively a URL rather than a platform path. We write all the paths in rst files with forward slashes. I suspect this means it shouldn't be doing `os.path.join()` either, but I didn't want to change more than I needed to at the moment.